libusb/os/darwin_usb.c

```diff
@@ -2276,8 +2276,11 @@ static int darwin_abort_transfers (struct usbi_transfer *itransfer) {
 
   usbi_dbg (ctx, "calling clear pipe stall to clear the data toggle bit");
 
-  /* newer versions of darwin support clearing additional bits on the device's endpoint */
+#if InterfaceVersion <= 245
+  /* with older releases of IOUSBFamily the OS always clears the host side data toggle. for
+     consistency also clear the data toggle on the device. */
```

Comment:
I think this difference should be documented in the description of `libusb_cancel_transfer`, with some text along the lines of:

> On Mac OS versions prior to 10.5, this function has the side effect of clearing the host-side data toggle bit, and will send a `ClearFeature(ENDPOINT_HALT)` request to clear the device-side data toggle. If the device does not implement this request correctly, the data toggle bits on the host and device may be left out of sync.

Reply:
This was done in #1121.